core/templates/dev/head/pages/admin/misc_tab/AdminMiscTabDirective.js

```diff
@@ -76,6 +79,40 @@ oppia.directive('adminMiscTab', [
         $scope.downloadTopicSimilaritiesFile = function() {
           $window.location.href = ADMIN_TOPICS_CSV_DOWNLOAD_HANDLER_URL;
         };
+
+        var setDataExtractionQueryStatusMessage = function(message) {
+          $scope.showDataExtractionQueryStatus = true;
+          $scope.dataExtractionQueryStatusMessage = message;
+        };
+
+        $scope.submitQuery = function() {
+          var STATUS_PENDING = (
+            'Data extraction query has been submitted. Please wait.');
+          var STATUS_FINISHED = 'Loading the extracted data ...';
+          var STATUS_FAILED = 'Error, ';
+
+          var data = {
+            exp_id: $scope.expId,
+            exp_version: $scope.expVersion,
+            state_name: $scope.stateName,
+            num_answers: $scope.numAnswers
+          };
+
+          setDataExtractionQueryStatusMessage(STATUS_PENDING);
+          var URL = DATA_EXTRACTION_QUERY_HANDLER_URL + '?';
+          angular.forEach(data, function(value, key) {
+            URL += key + '=' + value + '&';
```

Comment:
Also I'm not sure what the point of declaring the 'data' variable is; you might as well add the values directly.